### PR TITLE
[SYCL] Allow library mismatch for libdevice host object on Win32

### PR DESCRIPTION
**Pull request by jinge90**
VC++ yvals.h header file sets up linker directives like this:
  /FAILIFMISMATCH:RuntimeLibrary=MT_StaticRelease

This will break linking, if the final standard libraries linked
to the program have different configuration. Since the host part
of libdevice objects is empty, we do not care about the mismatch
checking.

Signed-off-by: jinge90 <ge.jin@intel.com>

### CODE REVIEW

File: libdevice/cmake/modules/SYCLLibdevice.cmake [hunk 1]
```diff
@@ -27,6 +27,10 @@ set(compile_opts
   -sycl-std=2017
   )
 
+if (WIN32)
+	set(compile_opts "-D_ALLOW_RUNTIME_LIBRARY_MISMATCH ${compile_opts}")
```
Comment (vzakhari) [inline libdevice/cmake/modules/SYCLLibdevice.cmake:31]: Looks like to many blanks.

Comment (vzakhari) [reply libdevice/cmake/modules/SYCLLibdevice.cmake:31]: Also, I would use `list(APPEND)`, because `compile_opts` is a list.

**Review by vzakhari — CHANGES_REQUESTED** [review]

File: libdevice/cmake/modules/SYCLLibdevice.cmake (hunk 1 above)
Comment (jinge90) [reply libdevice/cmake/modules/SYCLLibdevice.cmake]: Done.
Thanks very much.

**Comment by jinge90** [comment]
Hi, @vzakhari 
Could you help review the patch again?
Thanks very much.

**Review by vzakhari — APPROVED** [review]
Hi Ge, I am very sorry for such a huge delay.  I am not sure how it went under the radar.

**Pull request merged**